ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseLogEntry.java

```diff
@@ -43,27 +50,34 @@ public static ParseLogEntry of() {
       } catch (IOException e) {
         throw new UncheckedIOException(e);
       }
-      Optional.ofNullable(logEntry.path("insertId").textValue())
-          .ifPresent(id -> attributes.put(Attribute.DOCUMENT_ID,
-              UUID.nameUUIDFromBytes(id.getBytes(StandardCharsets.UTF_8)).toString()));
+      JsonNode fields = logEntry.path("jsonPayload").path("Fields");
+      String documentId = Optional.ofNullable(logEntry.path("insertId").textValue())
+          .map(id -> attributes.put(Attribute.DOCUMENT_ID,
+              UUID.nameUUIDFromBytes(id.getBytes(StandardCharsets.UTF_8)).toString()))
+          .orElse(UUID.randomUUID().toString());
+      attributes.put(Attribute.DOCUMENT_ID, documentId);
```

Comment:
if i understand correctly, this only needs to put `DOCUMENT_ID` once:
```suggestion
          .map(id -> UUID.nameUUIDFromBytes(id.getBytes(StandardCharsets.UTF_8)).toString())
          .orElse(UUID.randomUUID().toString());
      attributes.put(Attribute.DOCUMENT_ID, documentId);
```